From 82899b2219d9b1c67bc6528ab816f615dbee49a4 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Alex=20R=C3=B8nne=20Petersen?= <alex@alexrp.com>
Date: Tue, 16 Sep 2025 13:48:24 +0200
Subject: [PATCH 073/133] langref: runtime_shrExact_overflow does not work as
 expected on LoongArch
X-Developer-Signature: v=1; a=openpgp-sha256; l=914; i=xtex@astrafall.org;
 h=from:subject; bh=vApHKUsxcBfrLVgX0EmHBdl0B7Qs0BCMQ3rWkUfy33I=;
 b=owGbwMvMwCW2U4Ij7wZL9ETG02pJDJn78ljLmMxjX4R9rX8VNv/onn0xH643v5zPK6xwcN8Gi
 7k3/2je7ZjIwiDGxWAppshSZNjgzaqTzi+6rFwWZg4rE8gQaZEGBgYGBhYGvtzEvFIjHSM9U21D
 PUNDHSMdIwYuTgGY6mPnGP6Xi7+1Vdw7rX+XnEvja+15Jw8E5Gx4zbiLK3FGUNiqgv2ZjAzT4jq
 bwm8+1+QIuWivYr/xZsgqhhm3XziWFh6beMfURIATAA==
X-Developer-Key: i=xtex@astrafall.org; a=openpgp;
 fpr=7231804B052C670F15A6771DB918086ED8045B91

https://github.com/ziglang/zig/issues/24304
(cherry picked from commit c547a05b656118bef790478c26b439e07c858c76)
---
 doc/langref/runtime_shrExact_overflow.zig | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/doc/langref/runtime_shrExact_overflow.zig b/doc/langref/runtime_shrExact_overflow.zig
index 1350e9da75d4..98da0dde1191 100644
--- a/doc/langref/runtime_shrExact_overflow.zig
+++ b/doc/langref/runtime_shrExact_overflow.zig
@@ -7,7 +7,7 @@ pub fn main() void {
     const y = @shrExact(x, 2);
     std.debug.print("value: {}\n", .{y});
 
-    if (builtin.cpu.arch.isRISCV() and builtin.zig_backend == .stage2_llvm) @panic("https://github.com/ziglang/zig/issues/24304");
+    if ((builtin.cpu.arch.isRISCV() or builtin.cpu.arch.isLoongArch()) and builtin.zig_backend == .stage2_llvm) @panic("https://github.com/ziglang/zig/issues/24304");
 }
 
 // exe=fail
-- 
2.52.0

